kobuki_ws/src/kobuki_ros/kobuki_description/urdf/kobuki.urdf.xacro

```diff
@@ -221,7 +221,9 @@
     </link>
 
     <!-- Kobuki Gazebo simulation details -->
-    <xacro:kobuki_sim/>
+    <xacro:if value="$(arg is_sim)">
+        <xacro:kobuki_sim/>
+    </xacro:if>
```

Comment:
It is fine to remove these three lines if we only get data from a real robot. However, if we are using Gazebo for simulation, these lines are required for Gazebo configuration.

Additionally, if we are not opening Gazebo when using a real robot, it is okay to remove the if condition. The if condition is used to prevent misuse.

Reply:
Since I'm not familiar with Gazebo, is there a reference on this requirement? Thanks!

Reply:
I believe there isn't a comprehensive tutorial on setting up a robot in Gazebo with ROS 2. Many tutorials are written for ROS 1. Therefore, I frequently refer to the [ROS 1 documentation](https://classic.gazebosim.org/tutorials?tut=ros_gzplugins) and find corresponding [migration methods](https://github.com/ros-simulation/gazebo_ros_pkgs/wiki) for ROS 2.

For example, when configuring the differential drive plugin, I refer to the [documentation](https://classic.gazebosim.org/tutorials?tut=ros_gzplugins#DifferentialDrive) to understand the relevant parameters. Afterward, I check the [migration wiki](https://github.com/ros-simulation/gazebo_ros_pkgs/wiki/ROS-2-Migration:-Diff-drive) for differential drive to adapt it to ROS 2. While this approach works for me, it can be a bit cumbersome.

Reply:
Fortunately, the original configuration has already been migrated to ROS 2, 
and I only need to fix certain parts, such as the TF hierarchy.

Reply:
I'm not sure if I fully answered the question, you might be specifically interested in the tag `xacro:kobuki_sim`. 
This tag is utilized to include the Gazebo configuration. 
It is located in `kobuki_gazebo.urdf.xacro`, in the same folder as `kobuki.urdf.xacro`.

Reply:
Yes I'm interested in the `xacro:kobuki_sim` tag. I overlooked the use of it in `kobuki_gazebo.urdf.xacro`.

I also appreciate the additional documents and guidance on setting up the gazebo plugin!